Feature/set state

alia-peterson opened the pull request:
## *Purpose:*
*What does this merge do? Check all that apply.*
- [x] Adds a new feature
- [ ] Improves an existing feature
- [ ] Fixes a bug
- [ ] Cleans up (logs, tests, etc.)


## *Description: What does this PR do?*
- Update context to utilize API data
- Update navbar component and associated tests 
- Create jest unit tests


## *What are the relevant tickets?*
Closes #7, closes #17


## *Bugs:*
See Error Handling issue (#25)


## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have run Lighthouse/Wave checks for accessibility
- [x] New and existing unit tests pass locally with my change

camaragon committed:
Delete unnecessary api file

camaragon committed:
Build get and post requests

camaragon committed:
Restructure fetch requests with a throw error

camaragon committed:
Setup context with the fetch responses

camaragon committed:
Add a comment about useEffect

camaragon committed:
Make sure the CORS error is gone

camaragon committed:
Add .json() to the responses from fetch

camaragon committed:
Fetch get courses and modules

camaragon committed:
Add useEffect and update context

camaragon committed:
Add check context to circle chart

camaragon committed:
Refactor the useEffect for context

camaragon committed:
Add useContext for console.log()

camaragon committed:
Setup the utility file

srslie committed:
Merge branch 'main' into feature/setState

srslie committed:
Get course navbar functionality

Co-authored-by: Lucas Merchant <69216459+lbmerchant93@users.noreply.github.com>

srslie committed:
Add conditional for modules in NavBar, courseState

Co-authored-by: Alia Peterson <70297733+alia-peterson@users.noreply.github.com>
Co-authored-by: Lucas Merchant <69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon <69489633+camaragon@users.noreply.github.com>

alia-peterson committed:
Update context to include current course

Co-authored-by: Alice Ruppert <68885499+srslie@users.noreply.github.com>
Co-authored-by: Lucas Merchant 
<69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon 
<69489633+camaragon@users.noreply.github.com>

alia-peterson committed:
Update context in moduleDashboard

Co-authored-by: Alice Ruppert <68885499+srslie@users.noreply.github.com>
Co-authored-by: Lucas Merchant 
<69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon 
<69489633+camaragon@users.noreply.github.com>

alia-peterson committed:
Update returns

Co-authored-by: Alice Ruppert <68885499+srslie@users.noreply.github.com>
Co-authored-by: Lucas Merchant 
<69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon 
<69489633+camaragon@users.noreply.github.com>

alia-peterson committed:
Update courseDashboard and calculations for charts

Co-authored-by: Cameron Aragon 
<69489633+camaragon@users.noreply.github.com>

camaragon committed:
Install jest and babel-jest

Co-authored-by: Alice Ruppert <68885499+srslie@users.noreply.github.com>
Co-authored by: Alia Peterson <70297733+alia-peterson@users.noreply.github.com>
Co-authored-by: Lucas Merchant <69216459+lbmerchant93@users.noreply.github.com>

camaragon committed:
Add test data to the mock data

Co-authored-by: Alice Ruppert <68885499+srslie@users.noreply.github.com>
Co-authored by: Alia Peterson <70297733+alia-peterson@users.noreply.github.com>
Co-authored-by: Lucas Merchant <69216459+lbmerchant93@users.noreply.github.com>

camaragon committed:
Setup a jest testing file for utility calculations

Co-authored-by: Alice Ruppert <68885499+srslie@users.noreply.github.com>
Co-authored by: Alia Peterson <70297733+alia-peterson@users.noreply.github.com>
Co-authored-by: Lucas Merchant <69216459+lbmerchant93@users.noreply.github.com>

camaragon committed:
Add calc test data to the mock data

camaragon committed:
Add jest tests for the percentages by mod function

camaragon committed:
Add jest test data

camaragon committed:
Finish jest testing for calculations

alia-peterson committed:
Update navbar and tests

alia-peterson committed:
Merge branch 'feature/setState' of github.com:CourseChart/course-chart-fe into feature/setState

alia-peterson committed:
Update test files to pass cypress

Co-authored-by: Lucas Merchant 
<69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon 
<69489633+camaragon@users.noreply.github.com>

camaragon commented on .babelrc:
```diff
@@ -0,0 +1,3 @@
+{
+    "presets": ["next/babel"]
```
This file and code was necessary to get the jest tests to run properly. It seems to be a Next.js thing.

alia-peterson committed:
Add jest to packagejson for ci testing

Co-authored-by: Lucas Merchant 
<69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon 
<69489633+camaragon@users.noreply.github.com>

alia-peterson committed:
Update package file to run jest tests

Co-authored-by: Lucas Merchant 
<69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon 
<69489633+camaragon@users.noreply.github.com>

alia-peterson committed:
Fix yml job name

Co-authored-by: Alice Ruppert <68885499+srslie@users.noreply.github.com>
Co-authored-by: Lucas Merchant 
<69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon 
<69489633+camaragon@users.noreply.github.com>

alia-peterson committed:
Add utilities location to jest script

Co-authored-by: Lucas Merchant 
<69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon 
<69489633+camaragon@users.noreply.github.com>

alia-peterson committed:
Update test script

Co-authored-by: Lucas Merchant 
<69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon 
<69489633+camaragon@users.noreply.github.com>

alia-peterson committed:
Update test script

Co-authored-by: Lucas Merchant 
<69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon 
<69489633+camaragon@users.noreply.github.com>

alia-peterson committed:
Update test script to include npm run

alia-peterson committed:
Change test scripts again

alia-peterson committed:
Add an & to test script

alia-peterson committed:
Add cypress config info

lbmerchant93 commented on components/CircleChart.js:
```diff
@@ -0,0 +1,10 @@
+
+export default function checkContext() {
+  const { sharedState } = useContext()
+  console.log('dawg', sharedState)
```
missed a console.log here

camaragon replied:
Was just messing around with some data here, this whole file is just random code. It will be replaced with the actual CircleChart component in the next merge

alia-peterson committed:
Remove cypress run from test script

lbmerchant93 commented on components/NavBar.js:
```diff
@@ -5,33 +5,62 @@ import styles from '../styles/Navbar.module.scss'
         </Link>
-        <div className={styles.content}>
+
+        {(sharedState.currentCourse === course.id) &&
```
Great job on the conditional rendering here for the modules of a specific coures

lbmerchant93 commented on context/apiCalls.js:
```diff
@@ -0,0 +1,26 @@
+// For single course: endpoint = 'courses/:id'
+// For single module: endpoint = 'modules/:id'
+export const getData =  (endpoint) => {
+    // const settings = {mode: 'no-cors'}
```
If we dont need this anymore we should take this out

camaragon replied:
I figured we could clear out a lot of this commented code out during refactoring, but let's definitely take it out.

alia-peterson committed:
Undo change to test script

lbmerchant93 commented on context/app-context.js:
```diff
@@ -1,12 +1,47 @@
+                    courses: courses.data
+                })
+            })
+    }, []);
```
Great use of useEffect to set the sharedState

lbmerchant93 commented on context/app-context.js:
```diff
@@ -1,12 +1,47 @@
-import { createContext, useContext, useState } from 'react';
+import { createContext, useContext, useState, useEffect } from 'react';
 import mockData from './mock-data';
```
We can take out this mockData now that we are making fetch calls to the API

camaragon replied:
I have test data for the calculations in the mock data file, I'm going to try using fixtures instead. Although for now let's not take out the mockData file for now.

lbmerchant93 commented on cypress/integration/dashboard_spec.js:
```diff
@@ -1,40 +1,45 @@
+  const course1Api = 'https://course-chart-be.herokuapp.com/courses/1'
 
   beforeEach(() => {
+    cy.intercept('GET', coursesApi, { fixture: 'courses-api' })
```
Great job using fixtures and intercepting those fetch calls!

lbmerchant93 commented on package.json:
```diff
@@ -7,8 +7,10 @@
+    "cy:run": "cypress run --config video=false",
+    "jest:test": "jest",
+    "cy:test": "start-server-and-test start http://localhost:3000 cy:run",
+    "test": "npm run cy:test && npm run jest:test"
```
Continuous Integration while using Cypress and Jest is tricky! Glad we got that figured out though!

lbmerchant93 commented on pages/courseDashboard.js:
```diff
@@ -1,7 +1,37 @@
+  const [course, setCourse] = useState({})
+  const [activityTotals, setActivityTotals] = useState({})
+  const [percentageByMod, setPercentageByMod] = useState([])
+  const [percentageByActivity, setPercentageByActivity] = useState([])
```
I think having these multiple local states is awesome! I assume they will be passed down to the chart components through props?

alia-peterson replied:
Yes i think that was my intent. Hopefully that works out well, otherwise we can re-address

lbmerchant93 reviewed (approved):
Looks great!

camaragon commented on utilities/calculations.js:
```diff
@@ -0,0 +1,64 @@
+  }
+}
+
+    // getActivityMinTotal(activity) {
```
If Lucas doesn't need these functions for his chart data then we should be all good to delete these comments

alia-peterson replied:
Removed!

camaragon commented on utilities/calculations.test.js:
```diff
@@ -0,0 +1,46 @@
+describe("calculations", () => {
+  const allActivities = data.allActivities;
+  
+  //   it('should have a get percentages method', () => {
```
I left this as a comment because wasn't quite sure how to check if this is a function using jest and was curious if someone knows how.

alia-peterson committed:
Reverse order for test scripts

Co-authored-by: Alice Ruppert <68885499+srslie@users.noreply.github.com>
Co-authored-by: Lucas Merchant 
<69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon 
<69489633+camaragon@users.noreply.github.com>

camaragon commented on pages/moduleDashboard.js:
```diff
@@ -1,7 +1,22 @@
+  const [module, setModule] = useState({})
+
+  useEffect(() => {
+    console.log(sharedState);
```
Let's just be aware of this console.log() in the useEffect() here

alia-peterson replied:
Removed! I was using it for testing so we don't need it anymore.

alia-peterson committed:
Rework yml configuration file

alia-peterson committed:
Update config file again

Co-authored-by: Alice Ruppert <68885499+srslie@users.noreply.github.com>
Co-authored-by: Lucas Merchant 
<69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon 
<69489633+camaragon@users.noreply.github.com>

alia-peterson committed:
Update config file to include docker

Co-authored-by: Alice Ruppert <68885499+srslie@users.noreply.github.com>
Co-authored-by: Lucas Merchant 
<69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon 
<69489633+camaragon@users.noreply.github.com>

alia-peterson committed:
Change job names

Co-authored-by: Alice Ruppert <68885499+srslie@users.noreply.github.com>
Co-authored-by: Lucas Merchant 
<69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon 
<69489633+camaragon@users.noreply.github.com>

alia-peterson committed:
Add checkout to config file

Co-authored-by: Alice Ruppert <68885499+srslie@users.noreply.github.com>
Co-authored-by: Lucas Merchant 
<69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon 
<69489633+camaragon@users.noreply.github.com>

alia-peterson committed:
Add npm install and workflow to config

Co-authored-by: Alice Ruppert <68885499+srslie@users.noreply.github.com>
Co-authored-by: Lucas Merchant 
<69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon 
<69489633+camaragon@users.noreply.github.com>

alia-peterson committed:
Add npm install to test

Co-authored-by: Alice Ruppert <68885499+srslie@users.noreply.github.com>
Co-authored-by: Lucas Merchant 
<69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon 
<69489633+camaragon@users.noreply.github.com>

alia-peterson committed:
Consolidate jobs into one

Co-authored-by: Alice Ruppert <68885499+srslie@users.noreply.github.com>
Co-authored-by: Lucas Merchant 
<69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon 
<69489633+camaragon@users.noreply.github.com>

alia-peterson committed:
Update image tag 

Co-authored-by: Alice Ruppert <68885499+srslie@users.noreply.github.com>
Co-authored-by: Lucas Merchant 
<69216459+lbmerchant93@users.noreply.github.com>
Co-authored-by: Cameron Aragon 
<69489633+camaragon@users.noreply.github.com>

camaragon reviewed (approved):
This looks great y'all! Let's get these components hooked up.

srslie reviewed (approved):
TY
